docs/fsharp/style-guide/formatting.md

```diff
@@ -826,6 +837,32 @@ let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
             |> List.ofArray
             |> List.map (fun t -> t.GetMethods())
             |> Array.concat
+
+// Not OK either
+let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
+               |> List.ofArray
+               |> List.map (fun assm -> assm.GetTypes())
+               |> Array.concat
+               |> List.ofArray
+               |> List.map (fun t -> t.GetMethods())
+               |> Array.concat
+```
+
+This also applies to mutable setters:
+
+```fsharp
+// Preferred approach
+ctx.Response.Headers.[HeaderNames.ContentType]
+    <- Constants.jsonApiMediaType |> StringValues
+ctx.Response.Headers.[HeaderNames.ContentLength]
+    <- bytes.Length |> string |> StringValues
```

Comment:
I would much rather keep this consistent with e.g. `->` in pattern matching and keep the `<-` on the original line. I think it looks cleaner, and if the "body" spans multiple lines, you don't have to indent them an extra 3 spaces. Here's what I suggest:

```suggestion
ctx.Response.Headers.[HeaderNames.ContentType] <-
    Constants.jsonApiMediaType |> StringValues
ctx.Response.Headers.[HeaderNames.ContentLength] <-
    bytes.Length |> string |> StringValues
```

Here's what I meant by my last remark, which I don't want, and is a quasi-vanity-alignment (not sensitive to names, but still needs extra indentation due to the assignment operator, which is against the spirit of this PR):

```f#
ctx.Response.Headers.[HeaderNames.ContentType]
    <- some stuff
       |> that spans
       |> several lines
       |> and must be indented
ctx.Response.Headers.[HeaderNames.ContentLength]
    <- let x = 2
       x |> something |> somethingElse
```

Reply:
IMO you cannot really treat the `<-` operator in the same way as the `->` because they are completely different. In general, for the latter you find big amounts of code after it, so it's normal that most uses of it get separated with an EOL and add an indentation level. However, for the re-assignment operator I think it's a bad practice if you add a lot of statements because it kills readability.

Reply:
> In general, for the latter you find big amounts of code after it

I agree.

> so it's normal that most uses of it get separated with an EOL and add an indentation level

Your suggestion also has an EOL and add an indentation level. I just want to put the operator on the "left" side of the EOL.

> However, for the re-assignment operator I think it's a bad practice if you add a lot of statements because it kills readability.

I don't find that.

Everything in moderation, of course. My second example above with a separate `let` binding is pushing things, though I wouldn't rule it out outright. `<-` is similar to `=`, and it's no problem having complex expressions when declaring a value using `=`, and it may in many cases be preferable to declaring temporary values before the assignment. As for my first example, I certainly find the following perfectly readable and much prefer it to having `<-` force three extra spaces of indentation.

```f#
ctx.Response.Headers.[HeaderNames.ContentType] <-
    some stuff
    |> thatSpans
    |> severalLines
    |> andMustBeIndented
```

---

Apart from `<-` on the second line forcing extra indentation and thus seeming like a kind of "vanity indentation" whose removal is the point of this PR, I think my main issue is that formatting should be consistent with the initial `=` assignment, which of course is always on the first line. Even though it's valid syntax, I have never seen anyone write

```f#
let someValue
  = Constants.jsonApiMediaType
    |> StringValues
```

I think `<-` should be treated like `=` to be consistent with assignments and re-assignments:

```f#
let mutable someValue =
    Constants.jsonApiMediaType
    |> StringValues

someValue <- 
    Constants.jsonApiMediaType
    |> StringValues
```

Reply:
I mentioned above that it's more difficult to do manually. You have to indent more manually in the IDE.

Here's my suggestion. The IDE indents automatically.

![new](https://user-images.githubusercontent.com/7766733/99958709-2a19c200-2d8a-11eb-93c8-d8fcce9205bf.gif)

Here's the original PR suggestion. You have to indent manually both on the `<-` line and on the next line.

![old](https://user-images.githubusercontent.com/7766733/99958702-25eda480-2d8a-11eb-9fab-a6c069b48f5d.gif)

Reply:
I think I would prefer the suggestion to place `<-` on the same line largely due to the symmetry of `=` also almost always being on that same line.

Reply:
yeah @cmeeren convinced me on this one